src/test/java/seedu/address/model/person/PersonContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,89 @@
+        // same values -> returns true
+        PersonContainsKeywordsPredicate firstPredicateCopy = new PersonContainsKeywordsPredicate(
+                firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_personContainsKeywords_returnsTrue() {
+        // One keyword
+        PersonContainsKeywordsPredicate predicate = new PersonContainsKeywordsPredicate(
+                Collections.singletonList("Alice"));
+        assertTrue(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
+
+        // Multiple keywords
+        predicate = new PersonContainsKeywordsPredicate(Arrays.asList("Alice", "Bob"));
+        assertTrue(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
+
+        // Multiple keywords with tags
+        predicate = new PersonContainsKeywordsPredicate(Arrays.asList("Alice", "Bob", "intern", "tech"));
+        assertTrue(predicate.test(new PersonBuilder().withName(
+                "Alice Bob").withTags("Intern", "Tech").build()));
+
+        // Mixed-case keywords
+        predicate = new PersonContainsKeywordsPredicate(Arrays.asList("aLIce", "bOB"));
+        assertTrue(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
+    }
+
+    @Test
+    public void test_personDoesNotContainKeywords_returnsFalse() {
+        // Zero keywords
+        PersonContainsKeywordsPredicate predicate = new PersonContainsKeywordsPredicate(Collections.emptyList());
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice").build()));
+
+        // Only one matching keyword
+        predicate = new PersonContainsKeywordsPredicate(Arrays.asList("Bob", "Carol"));
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice Carol").build()));
+
+        // Non-matching keyword
+        predicate = new PersonContainsKeywordsPredicate(Arrays.asList("Carol"));
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
+
+        // Multiple keywords with tags, with non-matching tag
+        predicate = new PersonContainsKeywordsPredicate(Arrays.asList("Alice", "Bob", "intern", "tech"));
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice Bob").withTags("Intern").build()));
+
+        // Keywords match phone, email and address, but does not match name or tags
+        predicate = new PersonContainsKeywordsPredicate(
+                Arrays.asList("12345", "alice@email.com", "Main", "Street"));
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice").withPhone("12345")
+                .withEmail("alice@email.com").withAddress("Main Street").build()));
```

Comment:
I'm a little confused here. Should the predicate contain tags? Should the new person built contain different tags? Should the keyword predicate have a name at all?

If all is as it should be then feel free to merge.

Reply:
The Predicate here contains a bunch of non-name non-tag words, all of which are in other fields of the given person. Hope that helps.